### PR TITLE
Feature/rocket badges

### PR DESCRIPTION
**Pull request by Bupilipili**
### In this branch

- Rockets show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket" (as per design)

### CODE REVIEW

File: src/components/Rockets.js [hunk 1]
```diff
@@ -1,7 +1,54 @@
...
+                    className="Cbutton"
+                    onClick={() => handleCancelReservation(rocket.id)}
+                  >
+                    Cancel Reservation
+                  </button>
+                </div>
+              ) : (
+                <div>
+                  <p className="description">{rocket.description}</p>
+                  <button
```
Comment (Bhekisile) [inline src/components/Rockets.js:40]: Please fix the eslint error attached.
Screenshot (34)

File: src/redux/rockets/RocketSlice.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      const rocketId = action.payload;
+      return state.map((rocket) =>
+        rocket.id !== rocketId ? rocket : { ...rocket, reserved: !rocket.reserved }
+      );
+    },
+    cancelReservation: (state, action) => {
+      const rocketId = action.payload;
+      return state.map((rocket) =>
+        rocket.id !== rocketId ? rocket : { ...rocket, reserved: false }
+      );
```
Comment (Bhekisile) [inline src/redux/rockets/RocketSlice.js:18]: Same here.
Screenshot (34)

**Review by Bhekisile — CHANGES_REQUESTED** [review]
Hi @Bupilipili,

Good job so far!
There are some issues that you still need to work on but you are almost there!

File: src/redux/rockets/RocketSlice.js (hunk 2 above)
Comment (Bupilipili) [reply src/redux/rockets/RocketSlice.js:18]: Thanks for the review, 
But you see all the Linter checks have passed.

File: src/redux/rockets/RocketSlice.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  initialState: [],
+  reducers: {
+    setRockets: (state, action) => action.payload,
+    reserveRocket: (state, action) => {
+      const rocketId = action.payload;
+      return state.map((rocket) => (rocket.id !== rocketId ? rocket : { ...rocket, reserved: !rocket.reserved }));
+    },
+    cancelReservation: (state, action) => {
+      const rocketId = action.payload;
+      return state.map((rocket) => (rocket.id !== rocketId ? rocket : { ...rocket, reserved: false }));
```
Comment (Bhekisile) [inline src/redux/rockets/RocketSlice.js:14]: I can't open your app, please fix for UX.
Screenshot (40)

**Review by Bhekisile — CHANGES_REQUESTED** [review]
Hi @Bupilipili,

Good job so far!
There are some issues that you still need to work on but you are almost there!

File: src/redux/rockets/RocketSlice.js (hunk 3 above)
Comment (Bupilipili) [reply src/redux/rockets/RocketSlice.js]: Done, try again.

Comment (Bupilipili) [reply src/redux/rockets/RocketSlice.js]: Done 👍

**Review by Bhekisile — APPROVED** [review]
Hi @Bupilipili,

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Pull request merged**